### PR TITLE
Add support for Evo2 predict using a LoRA

### PR DESCRIPTION
**Pull request by balvisio**
### Description

Adds support for Evo2 predict using a LoRA checkpoint
### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Documentation update
- [ ] Other (please describe):

### CI Pipeline Configuration

Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

- If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
- If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
# TODO: Add code snippet
```

### Pre-submit Checklist

- [x] I have tested these changes locally
- [x] I have updated the documentation accordingly
- [x] I have added/updated tests as needed
- [ ] All existing tests pass successfully

## Summary by CodeRabbit

- **New Features**
  - Prediction: add CLI flag and Python API option to load a LoRA checkpoint for inference.
  - Training: add CLI options to configure LoRA hyperparameters (alpha and dimension) for fine-tuning.

- **Tests**
  - Add test helper to build prediction commands and an integration test comparing predictions with and without LoRA; a duplicate test entry was also introduced.

### CODE REVIEW

**Review by jwilber — APPROVED** [review]

**Comment by dorotat-nv** [comment]
@balvisio , evo2 finetune + lora does not work, see https://github.com/NVIDIA/bionemo-framework/issues/1136

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py [hunk 1]
```diff
@@ -16,13 +16,17 @@
...
 
 import pytest
+import torch
 
+from bionemo.core.data.load import load
+from bionemo.testing.data.fasta import ALU_SEQUENCE, create_fasta_file
 from bionemo.testing.subprocess_utils import run_command_in_subprocess
 
-from .common import small_training_cmd, small_training_finetune_cmd
+from .common import predict_cmd, small_training_cmd, small_training_finetune_cmd
```
Comment (dorotat-nv) [inline sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: it seems predict_cmd is not used

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py [hunk 2]
```diff
@@ -200,3 +204,83 @@ def test_train_evo2_finetune_runs(tmp_path, with_peft: bool):
             assert val_losses_ft[i][1] <= val_losses_ft[i - 1][1], (
                 f"Validation loss increased at step {val_losses_ft[i][0]}: {val_losses_ft[i][1]} > {val_losses_ft[i - 1][1]}"
             )
+
+@pytest.mark.timeout(512)
+@pytest.mark.slow
+def test_different_results_with_peft(tmp_path):
```
Comment (dorotat-nv) [inline sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: What are the different results?  Could the name be more specific? 
Also, what is the difference between this test and the previous one? Only that the checkpoint used is base_model_checkpoint_path, not the output of train?

File: sub-packages/bionemo-evo2/src/bionemo/evo2/run/predict.py
Comment (dorotat-nv) [inline sub-packages/bionemo-evo2/src/bionemo/evo2/run/predict.py:1]: predict unit test is missing. maybe predict can be tested in test_predict?

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py (hunk 1 above)
Comment (balvisio) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: Hi Dorota, are you looking at the right file / version? `predict_cmd` is being used in line 248 and 268

File: sub-packages/bionemo-evo2/src/bionemo/evo2/run/predict.py
Comment (balvisio) [reply sub-packages/bionemo-evo2/src/bionemo/evo2/run/predict.py:1]: I added a `predict` test in the context of fine-tuning. What `test_different_results_with_peft` does is:
1. calls `predict` using base model.
2. Finetunes a base model.
3. Calls `predict` using the fine-tuned model.
4. asserts that the output logits are different. 

Followed a pattern used for ESM2

I can create another predict unit test in a new file `test_predict.py` that does a sanity-check that `predict` works.

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py (hunk 2 above)
Comment (balvisio) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: What test_different_results_with_peft does is:

1. calls predict using base model.
2. Finetunes a base model.
3. Calls predict using the fine-tuned model. (Note that in line 268 we are using `base_model_checkpoint_path` but also passing `result_dir_peft` to load the LoRA weights)
4. Asserts that the output logits are different.

Followed a pattern used for ESM2

I can create another predict unit test in a new file `test_predict.py` that does a sanity-check that predict works.

Comment (dorotat-nv) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: Thanks for explaining! cant we move this test to test_predict and rename it test_predict_with_without_perft?

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py (hunk 1 above)
Comment (dorotat-nv) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: sorry! I didnt se it that time!

**Review by dorotat-nv — APPROVED** [review]

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py (hunk 2 above)
Comment (balvisio) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_finetune.py]: done
